plugins/cloudtrail/pkg/cloudtrail/source.go

```diff
@@ -248,6 +249,42 @@ func getMoreSQSFiles(pCtx *Plugin, oCtx *PluginInstance) error {
 		return fmt.Errorf("received SQS message that was not a SNS Notification")
 	}
 
+	if pCtx.Config.UseS3SNS {
+		// Process SNS message coming from S3
+		var (
+			s3Event    events.S3Event
+			s3Init     bool
+			lastBucket string
+		)
+
+		err = json.Unmarshal([]byte(sqsMsg["Message"].(string)), &s3Event)
+
+		if err != nil {
+			return err
+		}
+
+		for _, record := range s3Event.Records {
+
+			// init s3 and only re-init if bucket changes
+			if !s3Init || record.S3.Bucket.Name != lastBucket {
+				oCtx.s3.bucket = record.S3.Bucket.Name
+
+				initS3(oCtx)
```

Comment:
Do you think it's possible for `initS3` to be invoked multiple times? Is it safe?

Reply:
Looking through the function `initS3`, it initiates a new AWS session and creates a new downloader with this session. The intention behind this was as follows:
- The new event notification from S3 includes the [bucket](https://github.com/aws/aws-lambda-go/blob/main/events/s3.go#L40) details.
- If the bucket changes, set `oCtx.s3.bucket` to reflect the new bucket

So I believe its safe to initiate new sessions, but you are correct in identifying this - since I made the assumption that we'd need to init a new session to change the bucket, which is not the case, as we can simply re-use the same session to access the new bucket. Will updated the PR to reflect this change. Thank you!